spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/batch/SimpleJobService.java

```diff
@@ -242,10 +242,12 @@ public JobExecution stop(Long jobExecutionId) throws NoSuchJobExecutionException
 		if (!jobExecution.isRunning()) {
 			throw new JobExecutionNotRunningException("JobExecution is not running and therefore cannot be stopped");
 		}
-
+		// Indicate the execution should be stopped by setting it's status to
+		// 'STOPPING'. It is assumed that
+		// the step implementation will check this status at chunk boundaries.
 		logger.info("Stopping job execution: " + jobExecution);
 
-		jobExecution.setStatus(BatchStatus.STOPPED);
+		jobExecution.setStatus(BatchStatus.STOPPING);
 		jobRepository.update(jobExecution);
```

Comment:
> [!NOTE]
> This comment is around this general area (the stop method) and not necessarily this line....

> So when using Batch 4.x it was just a quick check of the JobRegistry to retrieve the job, which was always empty since SCDF never deals with Jobs directly. But with Batch 5.x they loaded the Job Registry and attempted to retrieve the Job in a different way using the SimpleJobOperator. 

It looks like `JsrJobOperator` [was using ](https://github.com/spring-projects/spring-batch/blob/fb0ce98a890eae1908bdf25d69eb4ea51b89973e/spring-batch-core/src/main/java/org/springframework/batch/core/jsr/launch/JsrJobOperator.java#L744)
```java
Job job = jobRegistry.getJob(jobExecution.getId());
```
and `SimpleJobOperator` [is using](https://github.com/spring-projects/spring-batch/blob/ffe3687c94412c3a9c156c1cc85e78bf974349b1/spring-batch-core/src/main/java/org/springframework/batch/core/launch/support/SimpleJobOperator.java#L340)
```java
Job job = jobRegistry.getJob(jobExecution.getJobInstance().getJobName());
```
so I'm guessing it is the job instance coming back null off the execution which is now causing the NPE. Correct?